code/modules/vehicles/mecha/combat/greyscale/greyscale_constructor.dm

```diff
@@ -173,12 +176,27 @@ GLOBAL_LIST_INIT(greyscale_weapons_data, generate_greyscale_weapons_data())
 	current_stats["slowdown"] = slowdown
 	current_stats["armor"] = armor
 
+/obj/machinery/computer/mech_builder/update_icon()
+	..()
+
+	// Cooldown
+	if(locked)
+		icon_state = "[initial(icon_state)]_cd"
+
+		if(machine_stat & (BROKEN|DISABLED))
+			icon_state += "b"
+
+		else if(machine_stat & NOPOWER)
+			icon_state += "0"
```

Comment:
use update icon state properly

Reply:
what do you mean by that

Reply:
OH THERE IS A SEPARATE PROC FOR UPDATE_ICON_STATE HERE, sorry didnt knew it existed